### ISSUE TITLE
[feature request] A better repr for panflute AST

### ISSUE THREAD

**Issue by ickc** [sergiocorreia/panflute#49]
For example,

```python
>>> from panflute import *
>>> table_referenced = Table(TableRow(TableCell(Plain(Str(''))), TableCell(Plain(Str('')))), TableRow(TableCell(Plain(Str('1'))), TableCell(Plain(Str('2')))), TableRow(TableCell(Plain(Str('3'))), TableCell(Plain(Str('4')))), alignment=['AlignDefault', 'AlignDefault'], width=[0.5, 0.5])
>>> repr(table_referenced)
"Table(TableRow(TableCell(Plain(Str())) TableCell(Plain(Str()))) TableRow(TableCell(Plain(Str(1))) TableCell(Plain(Str(2)))) TableRow(TableCell(Plain(Str(3))) TableCell(Plain(Str(4)))); alignment=['AlignDefault', 'AlignDefault'], width=[0.5, 0.5], rows=3, cols=2)"
```

It will be easier for writing tests if `repr(table_referenced)` would be just `Table(TableRow(TableCell(Plain(Str(''))), TableCell(Plain(Str('')))), TableRow(TableCell(Plain(Str('1'))), TableCell(Plain(Str('2')))), TableRow(TableCell(Plain(Str('3'))), TableCell(Plain(Str('4')))), alignment=['AlignDefault', 'AlignDefault'], width=[0.5, 0.5])`. i.e. if I copy the repr output and paste it, it would be a valid function to generate a panflute AST.

Currently, the `__str__` points to the `__repr__` method, may be the current repr will be moved to str, and has a new repr that has the said behavior?

**Comment by sergiocorreia** [comment]
Makes sense (also agrees with the [stated purpose](https://docs.python.org/3/library/functions.html#repr) of repr).

That said, it might be a lot of work to have valid repr for most objects. Will try to see if I can push a quick commit but not entirely sure of the timeline.

**Comment by sergiocorreia** [comment]
Might also be a good idea to add more convenience functions/attributes, such as testing for equality, or adding a startswith method.

See https://github.com/bmc/ebook-template/blob/master/scripts/pandoc-filter.py for ideas
